### PR TITLE
IA-3120 Hook up Azure relay and adapt to latest WSM APIs

### PR DESCRIPTION
**Pull request by Qi77Qi**
https://broadworkbench.atlassian.net/browse/IA-3120

* This PR also removes `sbt test` from jenkins job run since we already run all unit tests separate. This will reduce time to run leonardo-build from 17 min to about 7 min

* Remove public IP creation

Run azure automation test manully against fiab

---
Have you read [CONTRIBUTING.md](https://github.com/DataBiosphere/leonardo/blob/develop/CONTRIBUTING.md) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:

- [ ] I've documented my API changes in Swagger

In all cases:

- [ ] Get a thumbsworth of review and PO signoff if necessary
- [ ] Verify all tests go green (you can re-run automation tests with a comment saying `jenkins retest`
- [ ] Run the automation tests multiple times in parallel to weed out instability if applicable via a comment saying `multi-test`
- [ ] Squash and merge; Delete your branch after this
- [ ] Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

**Comment by Qi77Qi** [comment]
jenkins retest

**Comment by Qi77Qi** [comment]
jenkins retest

**Comment by Qi77Qi** [comment]
multi-test

**Comment by Qi77Qi** [comment]
jenkins retest

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/db/RuntimeControlledResourceComponent.scala [hunk 1]
```diff
@@ -52,10 +52,6 @@ object WsmResourceType {
     override def toString: String = "AZURE_VM"
   }
 
-  case object AzureIp extends WsmResourceType {
```
Comment (jdcanas) [inline http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/db/RuntimeControlledResourceComponent.scala:55]: Are we no longer creating the azure ip resource in azure via WSM? Does the relay proxy give us a different IP to the VM?

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/AzurePubsubHandler.scala [hunk 2]
```diff
@@ -7,44 +7,214 @@ import cats.effect.Async
...
+  }
+
+  private def getCommonFields(name: ControlledResourceName,
+                              resourceDesc: String,
+                              userEmail: WorkbenchEmail,
+                              resourceId: Option[WsmControlledResourceId]) =
+    ControlledResourceCommonFields(
+      name,
+      ControlledResourceDescription(resourceDesc),
+      CloningInstructions.Nothing, //TODO: these resources will not be cloned with clone-workspace. Is this correct?
```
Comment (jdcanas) [inline http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/AzurePubsubHandler.scala]: The comment seems to be correct given WSM common field documentation

File: http/src/test/resources/reference.conf [hunk 3]
```diff
@@ -308,6 +308,16 @@ azure {
            max-attempts = 120
            interval = 1 seconds
        }
+       runtime-defaults {
+        acr-credential {
+           username = "username"
```
Comment (jdcanas) [inline http/src/test/resources/reference.conf:313]: Are these actual credentials, or are they templated in via firecloud-develop?

**Review by jdcanas — APPROVED** [review]
LGTM, few comments inline

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/db/RuntimeControlledResourceComponent.scala (hunk 1 above)
Comment (Qi77Qi) [reply http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/db/RuntimeControlledResourceComponent.scala:55]: yep...we don't want to expose public IP for Azure VMs

File: http/src/test/resources/reference.conf (hunk 3 above)
Comment (Qi77Qi) [reply http/src/test/resources/reference.conf:313]: they're templated in firecloud-develop, which is why they only exist in [http/src/test/resources/reference.conf](https://github.com/DataBiosphere/leonardo/pull/2650/files/6b234b16d61f7d933d2d8f3bf94786068db4180e#diff-d7d5667e9c02913aa71ba812a20d79cf8999b9bf04584264d67e386adfb992b6) but not the main app reference.conf

**Pull request merged**